extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -604,25 +604,21 @@ export class AmpConsent extends AMP.BaseElement {
         const sourceBase = getSourceUrl(ampdoc.getUrl());
         const resolvedHref = resolveRelativeUrl(href, sourceBase);
         const xhrService = Services.xhrFor(this.win);
-        return ampdoc.whenFirstVisible().then(() => {
-          return expandConsentEndpointUrl(this.element, resolvedHref).then(
-            (expandedHref) => {
-              return xhrService.fetchJson(expandedHref, init).then((res) => {
-                try {
-                  return xhrService.xssiJson(
-                    res,
-                    this.consentConfig_['xssiPrefix']
-                  );
-                } catch (e) {
-                  userAssert(
-                    false,
-                    'Could not parse the `checkConsentHref` response.'
-                  );
-                }
-              });
-            }
-          );
-        });
+        return ampdoc.whenFirstVisible().then(() =>
+          expandConsentEndpointUrl(this.element, resolvedHref).then(
+            (expandedHref) =>
+              xhrService.fetchJson(expandedHref, init).then((res) =>
+                xhrService
+                  .xssiJson(res, this.consentConfig_['xssiPrefix'])
+                  .catch((e) => {
+                    user().error(
+                      'Could not parse the `checkConsentHref` response.',
+                      e
+                    );
+                  })
+              )
+          )
+        );
```

Comment:
Can't wait for this codebase to be able to use the async/await syntax. This example would be much more readable with it.

Reply:
Same! I would love nothing more than to be able to rewrite so much of the old-style Promise code we have. It's the source of like half of all production error reprorts